contracts/src/instances/reth/RETHBase.sol

```diff
@@ -0,0 +1,153 @@
+    /// @param _destination The destination of the withdrawal.
+    /// @return amountWithdrawn The amount of base withdrawn.
+    function _withdrawWithBase(
+        uint256 _shareAmount,
+        address _destination,
+        bytes calldata // unused
+    ) internal override returns (uint256 amountWithdrawn) {
+        // Burning rETH shares in exchange for ether.
+        // Ether proceeds are credited to this contract.
+        _rocketTokenReth.burn(_shareAmount);
+
+        // Amount of ETH that was withdrawn from the yield source and
+        // will be sent to the destination address.
+        amountWithdrawn = address(this).balance;
+
+        // Return withdrawn ether to the destination.
+        (bool success, ) = payable(_destination).call{ value: amountWithdrawn }(
+            ""
+        );
+        if (!success) {
+            revert IHyperdrive.TransferFailed();
+        }
+    }
+
+    /// @dev Process a withdrawal in vault shares and send the proceeds to the
+    ///      destination.
+    /// @param _shareAmount The amount of vault shares to withdraw.
+    /// @param _destination The destination of the withdrawal.
+    function _withdrawWithShares(
+        uint256 _shareAmount,
+        address _destination,
+        bytes calldata // unused
+    ) internal override {
+        // Transfer the rETH shares to the destination.
+        _rocketTokenReth.transfer(_destination, _shareAmount);
+    }
+
+    /// @dev Convert an amount of vault shares to an amount of base.
+    /// @param _shareAmount The vault shares amount.
+    /// @return baseAmount The base amount.
+    function _convertToBase(
+        uint256 _shareAmount
+    ) internal view override returns (uint256) {
+        return _rocketTokenReth.getEthValue(_shareAmount);
+    }
+
+    /// @dev Convert an amount of base to an amount of vault shares.
+    /// @param _baseAmount The base amount.
+    /// @return shareAmount The vault shares amount.
+    function _convertToShares(
+        uint256 _baseAmount
+    ) internal view override returns (uint256) {
+        return _rocketTokenReth.getRethValue(_baseAmount);
+    }
+
+    /// @dev Gets the total amount of base held by the pool.
+    /// @return baseAmount The total amount of base.
+    function _totalBase() internal pure override returns (uint256) {
+        // NOTE: Since ETH is the base token and can't be swept, we can safely
+        // return zero.
+        return 0;
+    }
+
+    /// @dev Gets the total amount of shares held by the pool in the yield
+    ///      source.
+    /// @return shareAmount The total amount of shares.
+    function _totalShares()
+        internal
+        view
+        override
+        returns (uint256 shareAmount)
+    {
+        return _rocketTokenReth.balanceOf(address(this));
+    }
+
+    /// @dev We override the message value check since this integration is
+    ///      payable.
+    function _checkMessageValue() internal pure override {}
```

Comment:
We should update this so that it the contract isn't payable since ETH isn't a deposit asset.

Reply:
This one needs to stay because the contract receives ether when we are withdrawing as base, aka burning reth -> ether

Reply:
We only use `_checkMessageValue` in `openLong`, `openShort`, `addLiquidity`, and `initialize`. This just prevents ETH from being sent by the user into Hyperdrive via one of these functions. It will not prevent ether from being received by the `receive` function. Let me know if I'm missing something, but if you revert to make this non-payable, all of the existing tests should still pass.

Reply:
Great explanation thank you! Yeah you are right I just needed to change what error I am expecting to revert.